### PR TITLE
Change plugins to use elasticsearch logging

### PR DESCRIPTION
**Pull request by thecoop**
Change various built-in plugins to use the elasticsearch logging API rather than log4j directly

### CODE REVIEW

**Comment by thecoop** [comment]
@pgomulka I'm guessing the BwC failure is because the new logging API is not present on v7?

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by thecoop** [comment]
@elasticsearchmachine update branch

File: plugins/discovery-gce/build.gradle [hunk 1]
```diff
@@ -23,7 +23,6 @@ dependencies {
   api "com.google.http-client:google-http-client-gson:${versions.google}"
   api 'com.google.code.findbugs:jsr305:3.0.2'
   api "commons-logging:commons-logging:${versions.commonslogging}"
-  api "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
```
Comment (pgomulka) [inline plugins/discovery-gce/build.gradle:26]: I wonder if this is safe? we probably don't use log4j-1.2 in this plugin, so this is required by some of its dependencies. We don't have bridges from log4j to es logging yet

**Review by pgomulka — CHANGES_REQUESTED** [review]

File: plugins/discovery-gce/build.gradle (hunk 1 above)
Comment (thecoop) [reply plugins/discovery-gce/build.gradle]: Hmm, not any more it seems. I've reverted this.

**Review by pgomulka — APPROVED** [review]
LGTM

**Pull request merged**